File: Lang-K/src/lang/lexer/Token.java
package lang.lexer;

public class Token {
	public Token(String string, TokenKind kind) {
		this.string = string;
		this.kind = kind;
	}
	
	public String string;
	public TokenKind kind;
	
	@Override
	public String toString() {
		return string + "\t: " + kind;
	}
}

File: src/main/scala/marg/token/Token.scala
package marg.token

class Token private(string: String, kind: TokenKind) {
  require(string != null)

  override def toString = string + " : " + kind.toString
  def isEOF = kind eq TokenKind.EOF
  def String = string
  def Kind = kind
}

object Token {
  def apply(string: String, kind: TokenKind) = new Token(string, kind)
  def apply(char: Char, kind: TokenKind) = new Token(char.toString, kind)

  val EOF = Token("", TokenKind.EOF)
}

